src/App.tsx

```diff
@@ -26,23 +27,40 @@ export default function App() {
         aria-label='forecast tabs'>
         <Tab
           label='Future forecast'
+          to='/future forecast'
+          component={Link}
           {...a11yProps(Tabs.FUTURE_FORECAST)}
         />
         <Tab
           label='Historical forecast'
+          to='/historical forecast'
+          component={Link}
           {...a11yProps(Tabs.HISTORICAL_FORECAST)}
         />
       </MUITabs>
-      <TabPanel
-        value={activeTab}
-        index={Tabs.FUTURE_FORECAST}>
-        {activeTab === Tabs.FUTURE_FORECAST && <FutureForecast />}
-      </TabPanel>
-      <TabPanel
-        value={activeTab}
-        index={Tabs.HISTORICAL_FORECAST}>
-        Historical Forecast
-      </TabPanel>
+
+      <Routes>
+        <Route
+          path='/future forecast'
+          element={
+            <TabPanel
+              value={activeTab}
+              index={Tabs.FUTURE_FORECAST}>
+              {activeTab === Tabs.FUTURE_FORECAST && <FutureForecast />}
```

Comment:
In this implementation we can get rid of state because contents of tab won't be rendered if another route is active